test/celeritas/grid/CubicSpline.test.cc

```diff
@@ -0,0 +1,370 @@
+        11.3782267757997,
+        11.3782267557938,
+        11.3782267555937,
+    };
+    {
+        // Values from scipy.interpolate.CubicSpline with bc_type='natural'
+        // Note: scipy returns a non-zero value for the first y'' (O(1e-6))
+        static double const expected_result[] = {
+            0,
+            129262831.4307167,
+            -4937289.102472972,
+            232278.9998184848,
+            -10781.66556976438,
+            500.6415331476759,
+            -23.24711350262595,
+            1.079470754955315,
+            -0.0501243277666431,
+            0.002326445592544685,
+            -0.0001057475269338493,
+            6.776263578034403e-21,
+        };
+        auto result
+            = SplineDerivCalculator(make_span(x), make_span(y), BC::natural)();
+        EXPECT_VEC_SOFT_EQ(expected_result, result);
+
+        SplineInterpolator interpolate({x[0], y[0], result[0]},
+                                       {x[1], y[1], result[1]});
+        EXPECT_SOFT_EQ(834.92850241567407, interpolate(1.1e-4));
+        EXPECT_SOFT_EQ(651.6053622151029, interpolate(5e-4));
+    }
+    {
+        // Values from scipy.interpolate.CubicSpline with bc_type='not-a-knot'
+        static double const expected_result[] = {
+            135721200.3235027,
+            122960668.4278613,
+            -4644650.528541619,
+            218690.4374550451,
+            -10150.68576333229,
+            471.342195341046,
+            -21.88655097124948,
+            1.016166966590067,
+            -0.04691224650074476,
+            0.001590245644093093,
+            0.00119268423306982,
+            -0.002782929877162913,
+        };
+        auto result = SplineDerivCalculator(
+            make_span(x), make_span(y), BC::not_a_knot)();
+        EXPECT_VEC_SOFT_EQ(expected_result, result);
+
+        SplineInterpolator interpolate({x[0], y[0], result[0]},
+                                       {x[1], y[1], result[1]});
+        EXPECT_SOFT_EQ(834.53755181860117, interpolate(1.1e-4));
+        EXPECT_SOFT_EQ(644.87140412068834, interpolate(5e-4));
+    }
+    {
+        static double const expected_result[] = {
+            122327823.66564,
+            110798659.11594,
+            -4492986.3810467,
+            211979.81140063,
+            -9839.4174867823,
+            456.88890796125,
+            -21.215420139232,
+            0.98500714700094,
+            -0.045473725878422,
+            0.0015414822331671,
+            0.0011561116748754,
+            -0.0026975939080425,
+        };
+        auto result
+            = SplineDerivCalculator(make_span(x), make_span(y), BC::geant)();
+        EXPECT_VEC_SOFT_EQ(expected_result, result);
+
+        SplineInterpolator interpolate({x[0], y[0], result[0]},
+                                       {x[1], y[1], result[1]});
+        EXPECT_SOFT_EQ(834.59530552174078, interpolate(1.1e-4));
+        EXPECT_SOFT_EQ(646.15145372907716, interpolate(5e-4));
+    }
+}
+
```

Comment:
I'd break the test in half here: before this is just `SplineDerivCalculator` (excellent tests by the way); and after you're testing the XS calculator. You could move those tests into the XsCalculator test itself.